pennylane/transforms/dynamic_one_shot.py

```diff
@@ -285,7 +285,7 @@ def gather_mcm(measurement, samples):
         meas_tmp = measurement.__class__(wires=wires)
         return meas_tmp.process_samples(mcm_samples, wire_order=wires)
     mcm_samples = np.array([mv.concretize(dct) for dct in samples]).reshape((-1, 1))
-    use_as_is = len(mv.measurements) == 1
+    use_as_is = mv.branches == {(0,): 0, (1,): 1}
```

Comment:
I like this, it even catches funky nested processing functions that end up implementing the identity :) 

Alternatively, one could also give the default processing function of `MeasurementValue` a name and check whether here whether it is still that one.

Reply:
Your counter-example showed I needed to match both keys and values :) Hopefully that will cover all cases.